src/scripts/ui/inline-roll-links.ts

```diff
@@ -289,23 +286,22 @@ export const InlineRollLinks = {
             if (item) rollLink.dataset.flavor ||= item.name;
         }
     },
+};
 
-    _documentFromDOM: (
-        html: HTMLElement,
-        foundryDoc?: ClientDocument
-    ): ActorPF2e | JournalEntry | JournalEntryPage<JournalEntry> | null => {
-        if (foundryDoc instanceof ChatMessagePF2e) return foundryDoc.actor ?? foundryDoc.journalEntry ?? null;
-        if (
-            foundryDoc instanceof ActorPF2e ||
-            foundryDoc instanceof JournalEntry ||
-            foundryDoc instanceof JournalEntryPage
-        ) {
-            return foundryDoc;
-        }
+/** If the provided document exists returns it, otherwise attempt to derive it from the sheet */
+function resolveDocument(html: HTMLElement, foundryDoc?: ClientDocument | null): ClientDocument | null {
+    if (foundryDoc) return foundryDoc;
 
-        const sheet: { id?: string; document?: unknown; actor?: unknown; journalEntry?: unknown } | null =
-            ui.windows[Number(html.closest<HTMLElement>(".app.sheet")?.dataset.appid)];
+    const sheet: { id?: string; document?: unknown } | null =
+        ui.windows[Number(html.closest<HTMLElement>(".app.sheet")?.dataset.appid)] ?? null;
 
-        return sheet.document instanceof ActorPF2e || sheet.document instanceof JournalEntry ? sheet.document : null;
-    },
-};
+    const document = sheet?.document;
+    return document && (document instanceof ActorPF2e || document instanceof JournalEntry) ? document : null;
```

Comment:
```suggestion
    return document instanceof ActorPF2e || document instanceof JournalEntry ? document : null;
```

Reply:
I should have scrutinized the original logic more closely than I did whoopsie